src/alfrescoApiClient.js

```diff
@@ -205,6 +205,11 @@ class AlfrescoApiClient extends ApiClient {
         // set header parameters
         request.set(this.defaultHeaders).set(this.normalizeParams(headerParams));
 
+        // set the proxy user name if present
+        if (!!proxyUserName) {
```

Comment:
Hi @bricka,

- I got your point that you need to use that param per request but the current behaviour of the alfresco-js-api is , you create a new AlfrescoApi object and you login.
Once you are login this instance identify a user. So if you need to impersonate another user you have to : 

-> logout and re-login 
or 
create another object. 

What I am trying to say is that proxyUserName has to come as parameter in the constructor and if you want change the user you have to provide or a method that change this proxyUserName or create another object wit different input parameter . 
With this solution you don't need to set the proxyUserName in any call. 
The limitation of your solution at the moment is h that it works  only with

 ``` getNodeChildren  ``` 

if we want spread it across all the method we need to modify all the methods firm and it's a really big change.

- Because this approach is valid only with the content service and we provide an interface also with the process service we should also think about how to document this thing or show an error if someone try to use it with the BPM